### ISSUE TITLE
[SDK3][Typescript] incorrect declaration for `FlowCardTriggerDevice#trigger()`

### ISSUE THREAD

**Issue by robertklep** [athombv/apps-sdk-issues#214]
`@types/homey/all.d.ts` declares `FlowCardTriggerDevice` as follows:
```
export class FlowCardTriggerDevice extends FlowCardTrigger {

}
```

However, `FlowCardTriggerDevice#trigger()` requires an additional argument (so cannot reuse the declaration of that method in `FlowCardTrigger`):
```
trigger(device: Device, tokens: object, state: object): Promise<any>;
```

**Comment by tjallingt** [comment]
Thanks for reporting this. I'm not sure why TypeScript doesn't allow this... This same issue affects `FlowCardTriggerDevice#getArgumentValues()` but that is "masked" by the error on `trigger()`.

I think the only fix is to extend `FlowCard` instead of `FlowCardTrigger` and then give `FlowCard#getArgumentValues()` a `device` argument with type `never`. Unfortunately I cannot find a way to prevent that from showing up in the JSDoc api reference...

**Comment by robertklep** [comment]
You could implement a base class for all common methods and implement the class-specific methods in their respective class implementations:
```
class FlowCardTriggerBase {
  …
}

class FlowCardTrigger extends FlowCardTriggerBase {
  trigger(tokens: object, state: object): Promise<any>;
  getArgumentValues(): Promise.<Array.<any>>;
}

class FlowCardTriggerDevice extends FlowCardTriggerBase {
  trigger(device: Device, tokens: object, state: object): Promise<any>;
  getArgumentValues(device: Device): Promise.<Array.<any>>;
}
```

**Comment by tjallingt** [comment]
Will release an update to the types tomorrow that fixes this (and makes `tokens` and `state` optional)

**Comment by tjallingt** [comment]
This is fixed in version `0.2.2` of `homey-apps-sdk-v3-types`

**Comment by fpalsson** [comment]
Great. Seems to work well.
